### PR TITLE
 Use common state for cloned view + Immutable set of paceholder salt + Improve fluent interface + Fix configs

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #188 

• Use common state for cloned instances of `View` and `WebView`.

• Add method `withClearedState()` for `View` and `WebView` that cleared state of view (parameters, blocks, etc.).

• Replace mutable method `setPlaceholderSalt()` of `ViewInterface` to immutable `withPlaceholderSalt()`.

• More methods of `View` and `WebView` returns self for improve fluent interface of classes.

• Fix configuration: set parameters after reset `View` and `WebView`.

• Improve changelog.

### CODE REVIEW

**Comment by samdark** [comment]
https://github.com/yiisoft/demo/pull/410

**Review by devanych — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -38,6 +38,57 @@ The package provides two use cases for managing view templates:
...
+- parameters,
+- blocks.
+
+State of `WebView` service:
+- parameters,
+- blocks,
+- title,
+- meta and link tags,
+- JS/CSS strings,
+- JS/CSS files.
```
Comment (samdark) [inline README.md:53]:
```suggestion
While being immutable and, by itself, stateless, both `View` and `WebView` services have sets of stateful and mutable data.

`View` service:
- parameters,
- blocks.

`WebView` service:
- parameters,
- blocks,
- title,
- meta and link tags,
- JS/CSS strings,
- JS/CSS files.
```

File: README.md [hunk 2]
```diff
@@ -38,6 +38,57 @@ The package provides two use cases for managing view templates:
...
+
+State of `View` and `WebView` services don't clone on clone of service instances, including when calling methods
+`with*()`. This allows use one state for cloned services. For example, we can get `WebView` from container in 
+controller via dependency injection and change context path:
```
Comment (samdark) [inline README.md:57]:
```suggestion
The state of `View` and `WebView` is not cloned when the services are cloned. So when
using `with*()`, both new and old instances are sharing the same set of stateful mutable data. It allows, for example, to get `WebView` via type-hinting in a controller and change context path:
```

File: README.md [hunk 3]
```diff
@@ -38,6 +38,57 @@ The package provides two use cases for managing view templates:
...
+```php
+final class BlogController {
+    private WebView $view;
+    public function __construct (WebView $view) {
+        $this->view = $view->withContextPath(__DIR__.'/views');
+    }
+}
+```
+
+... and then register CSS in widget:
```
Comment (samdark) [inline README.md:68]:
```suggestion
... and then register CSS in a widget:
```

File: README.md [hunk 4]
```diff
@@ -38,6 +38,57 @@ The package provides two use cases for managing view templates:
...
+    protected function run(): string
+    {
+        ...
+        $this->view->registerCss('.lastPosts { background: #f1f1f1; }');
+        ...
+    }
+}
+```
+
+For full clone of `View` or `WebView` service use method `withClearedState()` that also clear and clone state:
```
Comment (samdark) [inline README.md:86]:
```suggestion
To get a deep cloned `View` or `WebView` use `withClearedState()`: 
```

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**